src/Mvc/test/Mvc.FunctionalTests/ErrorPageTests.cs

```diff
@@ -1,27 +1,45 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Linq;
 using System.Net;
 using System.Net.Http;
 using System.Net.Http.Headers;
 using System.Text.Encodings.Web;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Mvc.Razor.Internal;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Testing;
 using Xunit;
+using Xunit.Abstractions;
 
 namespace Microsoft.AspNetCore.Mvc.FunctionalTests
 {
     /// <summary>
     /// Functional test to verify the error reporting of Razor compilation by diagnostic middleware.
     /// </summary>
-    public class ErrorPageTests : IClassFixture<MvcTestFixture<ErrorPageMiddlewareWebSite.Startup>>
+    public class ErrorPageTests : IClassFixture<MvcTestFixture<ErrorPageMiddlewareWebSite.Startup>>, IDisposable
     {
         private static readonly string PreserveCompilationContextMessage = HtmlEncoder.Default.Encode(
             "One or more compilation references are missing. Ensure that your project is referencing " +
             "'Microsoft.NET.Sdk.Web' and the 'PreserveCompilationContext' property is not set to false.");
-        public ErrorPageTests(MvcTestFixture<ErrorPageMiddlewareWebSite.Startup> fixture)
+        private readonly AssemblyTestLog _assemblyTestLog;
+
+        public ErrorPageTests(
+            MvcTestFixture<ErrorPageMiddlewareWebSite.Startup> fixture,
+            ITestOutputHelper testOutputHelper)
         {
-            Client = fixture.CreateDefaultClient();
+            _assemblyTestLog = AssemblyTestLog.ForAssembly(GetType().Assembly);
+
+            // var loggerProvider = new XunitLoggerProvider(testOutputHelper);
+            var loggerProvider = _assemblyTestLog.CreateLoggerFactory(testOutputHelper, GetType().Name);
+
+
+            var factory = fixture.Factories.FirstOrDefault() ?? fixture.WithWebHostBuilder(b => b.UseStartup<ErrorPageMiddlewareWebSite.Startup>())
```

Comment:
As much as I love terseness, could you break this down a bit? Make it a GetOrCreateLocal function and do
```
if(fixture.Factories.Any()){
return fixture.Factories.First()
}
else{
return fixture.WithWebHostBuilder(builder => {
    builder.UseStartup<...>()
         .ConfigureLogging(s => ....);
}
```